packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js

```diff
@@ -62,6 +87,22 @@ function useResolveEditedEntityAndContext( { postId, postType } ) {
 				postTypeToResolve,
 				postIdToResolve
 			) {
+				// For the front page, we alwayse use the front page template if existing.
+				if (
+					postTypeToResolve === 'page' &&
+					homepageId === postIdToResolve
+				) {
+					// We're still checking whether the front page template exists.
+					// Don't resolve the template yet.
+					if ( frontPageTemplateId === undefined ) {
+						return undefined;
+					}
+
+					if ( !! frontPageTemplateId ) {
+						return frontPageTemplateId;
+					}
```

Comment:
Are there any other values that `frontPageTemplateId` could be other than `undefined|string`? 

Could we just `return frontPageTemplateId;` without the two `if`s?

Reply:
Further up in the `useSelect` the value `false` is used to denote that no templates match the front page template. In this case, a page is set to be the front page, but there's no front page template, so these two if statements (and early returns) should be skipped, with the rest of this function continuing to execute to find the right template to use.

I _think_ that's what's happening, anyway!